pkg/controller/atlasfederatedauth/atlasfederatedauth.go

```diff
@@ -0,0 +1,100 @@
+package atlasfederatedauth
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"go.mongodb.org/atlas/mongodbatlas"
+
+	mdbv1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/workflow"
+)
+
+const (
+	ErrInvalidProvider    = "INVALID_PROVIDER"
+	ErrNotOrgGroupCreator = "NOT_ORG_GROUP_CREATOR"
+)
+
+func (r *AtlasFederatedAuthReconciler) ensureFederatedAuth(service *workflow.Context, fedauth *mdbv1.AtlasFederatedAuth) workflow.Result {
+	// If disabled, skip with no error
+	if fedauth.Spec.Enabled == nil || (fedauth.Spec.Enabled != nil && !*fedauth.Spec.Enabled) {
```

Comment:
This could be a method in the time directly.

```suggestion
	if !fedauth.Spec.IsEnabled() {
```